app/controllers/locations_controller.rb

```diff
@@ -90,13 +90,12 @@ def list_locations
     show_selected_locations(query, link_all_sorts: true)
   end
 
-  # Display list of locations that a given user is author on.
+  # Display list of locations that a given user created.
   def locations_by_user
-    user = if params[:id]
-             find_or_goto_index(User, params[:by_user].to_s)
-           else
-             @user
-           end
+    user = find_obj_or_goto_index(
+      model: User, obj_id: params[:by_user].to_s,
+      index_path: locations_path
+    )
```

Comment:
Changes functionality when `by_user` is not an existing user:
- in main: display locations of current user
- new: display all locations with a flash error